File: client/lib/models/transaction_split.dart
class TransactionSplit {
  final int id;
  final int subcategoryId;
  final double amount;
  final String? memo;
  final DateTime createdAt;

  TransactionSplit({
    required this.id,
    required this.subcategoryId,
    required this.amount,
    this.memo,
    required this.createdAt,
  });

  factory TransactionSplit.fromJson(Map<String, dynamic> json) {
    return TransactionSplit(
      id: json['id'],
      subcategoryId: json['subcategory_id'],
      amount: json['amount'].toDouble(),
      memo: json['memo'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}

File: server/app/models/transaction_split.py
from sqlalchemy import Column, Integer, Float, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.core.database import Base
from datetime import datetime


class TransactionSplit(Base):
    __tablename__ = "transaction_splits"
    id = Column(Integer, primary_key=True, index=True)

    transaction_id = Column(Integer, ForeignKey("transactions.id"), nullable=False, index=True)
    subcategory_id = Column(Integer, ForeignKey("subcategories.id"), nullable=False)
    amount = Column(Float, nullable=False)
    memo = Column(String, nullable=True)

    created_at = Column(DateTime, default=datetime.utcnow)

    # Relationships
    transaction = relationship("Transaction", back_populates="splits")
    subcategory = relationship("Subcategory")
